### ISSUE TITLE
E2guardian Updates Needed and E2guardian Internet Filtering Guide

### ISSUE THREAD

**Issue by Dalacor** [e2guardian/e2guardian#155]
Hi Fred and Phillip,

Not sure if this is the correct place to post this.

I  have created a complete setup guide using Slackware, Squid and E2Guardian in the hope that others will be able to benefit from my documentation as one of the biggest issues with E2guardian is the lack of a step by step setup guide.  While this guide is currently hosted on my website, I would recommend at some point in time that my guide be used on the E2guardian website where everyone can find it, although you will need to edit the guide as my guide has outstanding queries and issues.  In addition, I am not planning on making the guide on my website look good from a graphic design point of view as I don't have the time and the guide should be on your website anyway rather than a third party.

I have made quite a few changes to the default config as I have found in use that some default settings cause overblocking and others are simply not suitable for a school environment which is what most people I suspect are using the E2guardian for.

Secondly, I have several reccomendations for upgrading E2Guardian - if you look at my guide in the link posted, anything you see in red is either a question I have (as in how to do this) or I have highlighted parts of E2guardian that need to be updated as its broken - for example, in the E2guardian.conf file it is missing the authplugin ports as explained in my guide and there are other issues with E2guardian config files.  One of the biggest changes that need to be made is the inclusion of weapons, guns etc categories in the bannedsitelist file because this is necessary for UK Legislation for Schools and because there are so many categories in the blacklists now that are simply not there in the bannedsitelist file.

Lastly, I have highlighted in red questions that I have not as yet been able to answer or am working on.  It would be great to get all these questions answered in the final E2guardian Step by Step Guide so that everyone can benefit as I think that all too often people are asking the same questions over and over again.

Here is the link to the [E2Guardian Step by Step Guide](https://www.dalacor.co.uk/e2guardian.php)
I have also completely rewritten the template.html file and the link can be found on the E2Guardian Step by Step Guide or you can [view the template here.](https://www.dalacor.co.uk/template.html)

Please let me know what you think
Many thanks

**Comment by fredbcode** [comment]
Great ! Thanks, sorry I'm very busy now but I'll take a look of course
And unfortunately my English skills is not good enough

**Comment by Dalacor** [comment]
I appreciate that you are busy and I am still updating the guide anyway as I am putting in a completely updated system at one of my clients, so some of the stuff in red, will be resolved in the next few weeks anyway.  So there is no major rush.

Hopefully we will be able to take what I have done and provide a proper tutorial guide (which I really think is needed) that people can use to cover the basics as well as updating E2guardian to address some of the issues that I have raised.

**Comment by cnighswonger** [comment]
@Dalacor 

> authplugin = '/usr/local/e2guardian/etc/e2guardian/authplugins/port.conf' This is missing as well. Confirm with Fred whether all these things should be added or not. I have added it to my file. 

authplugins are correctly populated (now) when building from the latest master branch.

> Note: It appears that there are quite a few new blacklists folders, but are not included in the Bannedsitelist file. Presumably the same with Phraselists file etc. Review this and report to Fredb as categories like weapons etc should be included in the bannedsitelist file.

There are many blacklists available on the internet other than those from Squidguard. They all tend to have their own categories. It is a trivial matter to write a shell script to correctly create the .Include lines for each. There are also a number of blacklist "update" scripts available to help keep your lists current.

**Comment by fredbcode** [comment]
Sorry dalacor I have no time right now, please can you write some words here https://groups.google.com/forum/#!forum/e2guardian ?

**Comment by fredbcode** [comment]
I will stick your link in front of forum

**Comment by spikedrba** [comment]
Hi @Dalacor , thank you very much for putting together a very comprehensive guide. I'm trying to help out with the wiki and would love to integrate your guide into it if that's ok.

In that page I'd like to stick to just installing E2G since that's the topic here, ie not include how to install slackware, and then link your site/page as original source and point the availability of extended info there.

Is that ok with you? Is there any chance that you have the doc in markdown? that'd be extremely helpful to speed things up.

thanks,

Spike

**Comment by Dalacor** [comment]
@cnighswonger 

Sorry I have taken so long to get back to you, been so busy here!  When I do my next upgrade, I will confirm about authplugins, but if memory serves me correctly, I had to create the actual file and create a link in the e2guardian.config file.

Yes I have worked out that the problem is that the default blacklist which is supplied by URL Blacklists is not comprehensive enough and actually there is some concern that they are coasting on their reputation and not really updating anything anymore.  I will be reviewing different blacklists this week or next as I need to roll out more machines next month.

**Comment by Dalacor** [comment]
@spikedrba 

You are more than welcome to use the information that I have provided to create a wiki.  Unfortunately I only have the information in word and as you see it on the web page.  I did not bother with graphic design etc for the website page as I didn't want to spend a lot of time on the documentation as I am rather busy.  I have been meaning to get back to the webpage for weeks, but still have not got around to it.

I would suggest waiting a few weeks, as I am going to be rolling out a few more machines in April so I will install the latest version (I believe that version 4 is out), so will update the documentation to reflect the changes as well as update the documentation in areas where I have stuff in red and have worked out what to do.  That is assuming that there are any changes that I myself need to make for version 4

Having said that, would a wiki be of help as we don't want dozens of possibly inaccurate guides all over the internet as some of my information may be wrong as I may be misunderstanding some things etc.  Long term I think the best people to create the guides is the developers who understand their product.

**Comment by spikedrba** [comment]
@Dalacor v4 is not out for the general public and will prolly be a while before it's stable, just out for beta testing for now.

I've linked your page, but yeah, I share the same sentiment about documenting everything on the official wiki here on github (if you take a look you'll see much have changed and there's quite a few updates). If you can attach your word doc that may make the copy paste a bit faster than from the html page.

thanks,

**Comment by spikedrba** [comment]
The page you wrote has been linked and several others have been added about installation and initial configuration of e2g. Closing this issue, please reopen if you think something is missing. thanks.

**Comment by Dalacor** [comment]
I can see two issues just from a brief viewing.

You need to put on the homepage (https://github.com/e2guardian/e2guardian) that the Wiki is where you will find instructions on installing and using E2guardian.  Most people like myself are not that familiar with Github and I don't believe that I have ever visited that link before. Most people will never find the instructions as they won't know to look there.

Secondly the Install on Slackware Link on this page - https://github.com/e2guardian/e2guardian/wiki/Installation - goes to install from source not the same page as Install on Slackware in the box on the RHS

Third, I would put Install on Slackware after Install From Source in the box on the RHS. It's current location in your menu is not logical - it logically goes with Install from Ubuntu and Install from source

**Comment by spikedrba** [comment]
@Dalacor , thanks for catching that, 2 and 3 fixed, I had the name wrong which also meant the sorting order on the right bar was off. @fredbcode can you add a pointer to the wiki on the readme or do you want me to? have not made changes to the repo itself.

**Comment by Dalacor** [comment]
Yes I think it might be a good idea to include a link to the documentation in the install or readme file as most people will read the install file when installing the program or when they are looking for more information.  When I have time I will review what you have written up and offer my feedback. But I won't have time for a few weeks.
